server/app/models/grid_service.rb

```diff
@@ -92,7 +94,10 @@ def default_stack?
 
   # @param [String] state
   def set_state(state)
-    self.set(:state => state)
+    if self.state != state
```

Comment:
Is this necessarily safe with multiple master processes doing work at the same time?